app/questionnaire/questionnaire_schema.py

```diff
@@ -366,17 +366,14 @@ def is_answer_in_repeating_section(self, answer_id: str) -> Optional[bool]:
         if block := self.get_block_for_answer_id(answer_id):
             return self.is_block_in_repeating_section(block_id=block["id"])
 
-    def get_list_item_id_for_answer_id(
-        self, answer_id: str, list_item_id: Optional[str]
-    ) -> Optional[str]:
-        if (
-            list_item_id
-            and not self.is_answer_in_list_collector_block(answer_id)
-            and not self.is_answer_in_repeating_section(answer_id)
-        ):
-            return None
-
-        return list_item_id
+    def answer_should_have_list_item_id(
```

Comment:
To me `has` implies that given an answer, tell me if it has a list item id, which is not what this method is doing. This is telling me, given an answer, whether it should have a list item id based on whether it can have more than one instance. Happy to rename it to `should_answer_have_list_item_id` which I think is clearer. Thoughts?

Reply:
Ok, either is fine then

Reply:
Renamed to `should_answer_have_list_item_id` as I think it is clearer.

Reply:
How about `is_repeating_answer`?

Reply:
The only reason I initially swayed away from that was the func calls both `is_answer_in_list_collector_block` and `is_answer_in_repeating_section` so thought it would be confusing to call it `repeating answer` as it might not be obvious list collector block answers could also have list item ids. I don't really mind, I am happy to go with `is_repeating_answer`. @LJBabbage happy for me to change?

Reply:
Fine by me

Reply:
Updated to `is_repeating_answer`.